### PR TITLE
chore(validation): schema to request

### PR DESCRIPTION
**Pull request by abuaboud**

### CODE REVIEW

File: packages/backend/src/flows/flow-version/flow-version.service.ts [hunk 1]
```diff
@@ -159,24 +161,45 @@ function validateTrigger(settings: PieceTriggerSettings) {
...
+        propsSchema[name] = Type.Number({});
+        break;
+      case PropertyType.DROPDOWN:
+        propsSchema[name] = Type.Any({});
+        break;
+      case PropertyType.OAUTH2:
+        // Only accepts connections variable.
+        propsSchema[name] = Type.RegEx(RegExp('\$\{connections.(.*?)\}', 'g'));
+        break;
+    }
```
Comment (khaledmashaly) [inline packages/backend/src/flows/flow-version/flow-version.service.ts:196]: There's `SECRET_TEXT`, and `CUSTOM_AUTH` now?

File: packages/shared/src/collections/dto/update-collection-request.ts [hunk 2]
```diff
@@ -1,17 +1,50 @@
...
+        settings: Type.Object({}),
+        value: Type.Object({}),
+    }),
+    Type.Object({
+        key: Type.String(),
+        type: Type.Literal(ConfigType.OAUTH2),
+        settings: Type.Object({}),
+        value: Type.Object({}),
+    })
+
```
Comment (khaledmashaly) [inline packages/shared/src/collections/dto/update-collection-request.ts:42]: This comment is irrelevant to this PR, but do we need to add a `ConfigType` for `CUSTOM_AUTH`?

File: packages/backend/src/flows/flow-version/flow-version.service.ts (hunk 1 above)
Comment (abuaboud) [reply packages/backend/src/flows/flow-version/flow-version.service.ts:196]: It doesn't matter, The frontend should always input the reference `${connections.NAME}`

File: packages/shared/src/collections/dto/update-collection-request.ts (hunk 2 above)
Comment (abuaboud) [reply packages/shared/src/collections/dto/update-collection-request.ts:42]: Nope I will remove Authentication configs once I merged app-auth branch, but i couldn't make this branch compiles without them

**Review by khaledmashaly — APPROVED** [review]

**Pull request merged**